src/main/java/orlov/programming/spring_core_gym/dao/impl/training/TrainingDAO.java

```diff
@@ -0,0 +1,88 @@
+package orlov.programming.spring_core_gym.dao.impl.training;
+
+import lombok.extern.log4j.Log4j2;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+import orlov.programming.spring_core_gym.dao.DAO;
+import orlov.programming.spring_core_gym.dao.DAOSelectableCreatable;
+import orlov.programming.spring_core_gym.model.training.Training;
+import orlov.programming.spring_core_gym.model.user.Trainee;
+import orlov.programming.spring_core_gym.model.user.Trainer;
+import orlov.programming.spring_core_gym.storage.Storage;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+@Log4j2
+@Repository
+public class TrainingDAO implements DAOSelectableCreatable<Training> {
```

Comment:
Why not create a contract just for TrainingDao functionality?

Reply:
But DAOSelectableCreatable is already a contract for all Training operations.